### ISSUE TITLE
Gradle Application Plugin Main Class Not Recognized

### ISSUE THREAD

**Issue by jamesward** [GoogleContainerTools/jib#1501]
**Description of the issue**:
I'm using the Gradle application plugin and setting the main class for that:
```
application {
    mainClassName = "WebAppKt"
}
```

But the jib plugin is not picking up that setting:
```
com.google.cloud.tools.jib.plugins.common.MainClassInferenceException: Main class was not found, perhaps you should add a `mainClass` configuration to jib
```

**Expected behavior**:
The main class should be picked up for jib if set for the application plugin.

**Steps to reproduce**:
Create a `build.gradle.kts` with:
```
plugins {
    java
    application
    kotlin("jvm") version "1.3.20"
    id("kotlinx-serialization") version "1.3.20"
    id("com.google.cloud.tools.jib") version "1.0.0"
    id("com.github.johnrengelman.shadow") version "4.0.4"
}

repositories {
    mavenLocal()
    mavenCentral()
    maven("https://kotlin.bintray.com/kotlinx")
}

dependencies {
    compile("io.ktor:ktor-server-core:1.1.3")
    compile("io.ktor:ktor-server-netty:1.1.3")
    compile("io.ktor:ktor-freemarker:1.1.3")
    compile("io.ktor:ktor-client-cio:1.1.3")
    compile("io.ktor:ktor-client-json-jvm:1.1.3")
    compile("ch.qos.logback:logback-classic:1.2.3")
}

java {
    sourceCompatibility = JavaVersion.VERSION_1_8
}

tasks.compileKotlin {
    kotlinOptions.jvmTarget = "1.8"
}

application {
    mainClassName = "WebAppKt"
}

jib {
    val projectId: String? by project
    to.image = "gcr.io/$projectId/where-am-i"
}
```

Run with the latest gradle.

**Environment**: Linux

**Comment by chanseokoh** [comment]
Hi @jamesward,

IIRC, Jib is not able to automatically infer a main class in Kotlin (but I need to double-check that). I'm sure setting it explicitly will work.
```
jib {
  ...
  container.mainClass = "WebAppKt"
}
```
That said, we should understand why we cannot infer it for Kotlin.

**Comment by jamesward** [comment]
Jib should have access to the setting used by the application plugin and could use that if it exists.  Not really anything to do with inference or Kotlin.

**Comment by chanseokoh** [comment]
Ah, yeah, that's true. Would be good to check that info. But I still wonder why Jib cannot infer it, given that it searches through all `.class` files.

**Comment by loosebazooka** [comment]
Hey @jamesward : is this fix by any chance relevant? https://github.com/GoogleContainerTools/jib/pull/1457

Also, I think it can make sense to add support to reading application plugin config to jib?

**Comment by jamesward** [comment]
That fix might handle the auto-detection but really what I'm after is just reuse of the application plugin's setting as a default.

**Comment by chanseokoh** [comment]
The auto-detection is fixed in 1.0.1, and #1503 is opened for reusing the application plugin's settings. I'll close this issue.
